torch/_functorch/_aot_autograd/autograd_cache.py

```diff
@@ -274,6 +276,11 @@ def autograd_cache_key(
     Generate a unique hash of the FX graph for caching.
     """
     check_cacheable(gm)
+    if not torch._dynamo.config.specialize_float:
+        # TODO: remove this once specializing floats no longer changes
+        # the output of AOTAutograd, or when float guards are added to
+        # FXGraphCache
```

Comment:
hmm my understanding was that @bobrenjc93 's [PR](https://github.com/pytorch/pytorch/pull/140346) to restart analysis whenever we specialize a float should make that logic work with cache? Or is there another reason you think it's worth waiting till float guards land

Reply:
Yeah @jamesjwu and I discussed it this morning. This should no longer be necessary and should be removed.